### PR TITLE
Mock related tags tests

### PR DESCRIPTION
**Pull request by pmareke**
EX-4470

### CODE REVIEW

File: packages/x-components/tests/e2e/cucumber/common-steps.spec.ts [hunk 1]
```diff
@@ -101,3 +102,11 @@ Given('a suggestions API', () => {
     });
   });
 });
+
+Given('a related tags API', () => {
+  cy.intercept('https://api.empathy.co/getRelatedTags', req => {
+    req.reply({
+      relatedTags: getRelatedTagsStub()
+    });
+  }).as('interceptedRelatedTags');
```
Comment (CachedaCodes) [inline packages/x-components/tests/e2e/cucumber/common-steps.spec.ts:111]: Interceptions from the `common-steps` shouldn't need to be checked by the tests as they are generic. No need to make an alias for them.

```suggestion
  });
```

- [ ] Remove alias

File: packages/x-components/tests/e2e/cucumber/related-tags/related-tags.spec.ts [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
+Given('a related tags API with a selected one', () => {
+  cy.intercept('https://api.empathy.co/getRelatedTags', req => {
+    req.reply({
+      relatedTags: [
+        createRelatedTagStub('lego', 'bombero', false, ''),
+        createRelatedTagStub('lego', 'policia', false, 'bombero'),
+        createRelatedTagStub('lego', 'barbie', true, 'policia')
+      ]
+    });
+  }).as('interceptedRelatedTags');
```
Comment (CachedaCodes) [inline packages/x-components/tests/e2e/cucumber/related-tags/related-tags.spec.ts:29]: You're giving the same alias to both of the `intercept`'s so the second one is overriding the first, give the second one a different name.

```suggestion
  }).as('interceptedRelatedTagsWithSelection'); // Quick suggestion
```

- [ ] Give a new alias

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: packages/x-components/tests/e2e/cucumber/common-steps.spec.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
-import { Given, Then, When } from 'cypress-cucumber-preprocessor/steps';
 import { PageableRequest } from '@empathyco/x-adapter';
+import { Given, Then, When } from 'cypress-cucumber-preprocessor/steps';
 import { getNextQueriesStub } from '../../../src/__stubs__/next-queries-stubs.factory';
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/tests/e2e/cucumber/common-steps.spec.ts:3]: Sure!

File: packages/x-components/src/__stubs__/index.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export * from './banners-stubs.factory';
```
Comment (davidmfempathy) [inline packages/x-components/src/__stubs__/index.ts:1]: Why do we need this barrier file?

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 5]
```diff
@@ -43,14 +43,14 @@ describe('testing next queries component', () => {
     const wrapperComponent = {
       template: `
         <NextQueries>
-          <template #suggestion-content="suggestionContentScope">
-            <img src="./next-query-icon.svg" class="x-next-query__icon" data-test="icon"/>
-            <span
+        <template #suggestion-content="suggestionContentScope">
```
Comment (davidmfempathy) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:46]: Review the indentation of this file, I think it was fine before.

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 6]
```diff
@@ -77,21 +77,21 @@ describe('testing next queries component', () => {
     const wrapperComponent = {
       template: `
         <NextQueries>
-          <template #suggestion="suggestionScope">
```
Comment (davidmfempathy) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:80]: The same here.

File: packages/x-components/tests/e2e/cucumber/history-queries.feature [hunk 7]
```diff
@@ -15,9 +16,9 @@ Feature: History queries component
     And   next queries are displayed
     And   related tags are displayed
     Examples:
-      | hideIfEqualsQuery | debounceInMs | maxItemsToStore | maxItemsToRender | instant  | historyQueryItem | list                                                                                                           |
-      | true              | 150          | 15              | 8                | true     | 7                | puzzle, funko, puzzle 3d, funko pop, puzzle 3d harry, funko pop iron, puzzle 3d harry potter, funko pop iron man |
-      | false             | 150          | 15              | 8                | true     | 6                | puzzle, funko, puzzle 3d, funko pop, puzzle 3d harry, funko pop iron, puzzle 3d harry potter, funko pop iron man |
+      | hideIfEqualsQuery | debounceInMs | maxItemsToStore | maxItemsToRender | instant | historyQueryItem | list                                                                                                             |
```
Comment (davidmfempathy) [inline packages/x-components/tests/e2e/cucumber/history-queries.feature:19]: Remove the extra space at the end of the word list.

**Review by davidmfempathy — CHANGES_REQUESTED** [review]

File: packages/x-components/tests/e2e/cucumber/history-queries.feature (hunk 7 above)
Comment (pmareke) [reply packages/x-components/tests/e2e/cucumber/history-queries.feature:19]: What do you mean? I can't see the error here

image

File: packages/x-components/src/__stubs__/index.ts (hunk 4 above)
Comment (CachedaCodes) [inline packages/x-components/src/__stubs__/index.ts:1]: We still need to update all the imports that use functions from the `__stubs__` folder so they use this new index file. We can make it here or in a new task (my vote is for a new task)

Comment (pmareke) [reply packages/x-components/src/__stubs__/index.ts:1]: Yes I think so.

Comment (pmareke) [reply packages/x-components/src/__stubs__/index.ts:1]: I'll create the task.

Comment (pmareke) [reply packages/x-components/src/__stubs__/index.ts:1]: https://searchbroker.atlassian.net/browse/EX-4590

File: packages/x-components/src/__stubs__/related-tags-stubs.factory.ts [hunk 8]
```diff
@@ -37,13 +21,31 @@ export function getRelatedTagsStub(): RelatedTag[] {
...
+ * @param tag - The related tag.
+ * @param selected - An boolean if the RT is selected.
+ * @param relatedTag - An optional object with fields to override the related tag.
+ *
+ * @returns A related tag.
+ */
+export function createRelatedTagStub(
+  query: string,
+  tag: string,
+  selected = false,
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/__stubs__/related-tags-stubs.factory.ts:40]: I'm not sure about this one as if you try the following code

`createRelatedTagStub('Related Tag Query', `Related Tag`, { selected: true })`

You have an error regarding some missing properties of the RelatedTag model

image

[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/__stubs__/related-tags-stubs.factory.ts:40]: I like it!

**Review by CachedaCodes — APPROVED** [review]

**Review by davidmfempathy — APPROVED** [review]

**Pull request merged**